cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala

```diff
@@ -71,6 +75,37 @@ abstract class DrsPathResolver(drsConfig: DrsConfig) {
   def resolveDrsThroughMartha(drsPath: String, fields: NonEmptyList[MarthaField.Value]): IO[MarthaResponse] = {
     rawMarthaResponse(drsPath, fields).use(httpResponseToMarthaResponse(drsPathForDebugging = drsPath))
   }
+
+  def openChannel(accessUrl: AccessUrl): IO[ReadableByteChannel] = {
+    IO {
+      val httpGet = new HttpGet(accessUrl.url)
+      accessUrl.headers.getOrElse(Map.empty).toList foreach {
+        case (name, value) => httpGet.addHeader(name, value)
+      }
+      val client = httpClientBuilder.build()
+      val response = client.execute(httpGet)
+      val inner = Channels.newChannel(response.getEntity.getContent)
+      new ReadableByteChannel {
```

Comment:
perhaps a comment here that this RBC is here to manage the `close` for the other three resources